frontend/src/pages/FunctionalComponent/RecurringRunDetailsV2FC.tsx

```diff
@@ -0,0 +1,207 @@
+      try {
+        return await Apis.recurringRunServiceApi.getRecurringRun(recurringRunId);
+      } catch (err) {
+        const errorMessage = await errorToMessage(err);
+        updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message:
+            `Error: failed to retrieve recurring run: ${recurringRunId}.` +
+            (errorMessage ? ' Click Details for more information.' : ''),
+          mode: 'error',
+        });
+        return;
+      }
+    },
+    { enabled: !!recurringRunId, staleTime: 0, cacheTime: 0 },
+  );
+
+  const experimentId = recurringRun?.experiment_id!;
+  const { data: experiment } = useQuery<V2beta1Experiment | undefined, Error>(
+    ['experiment'],
+    async () => {
+      try {
+        return await Apis.experimentServiceApiV2.getExperiment(experimentId);
+      } catch (err) {
+        const errorMessage = await errorToMessage(err);
+        updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message:
+            `Error: failed to retrieve this recurring run's experiment.` +
+            (errorMessage ? ' Click Details for more information.' : ''),
+          mode: 'warning',
+        });
+        return;
+      }
+    },
+    { enabled: !!experimentId, staleTime: 0 },
+  );
+
+  useEffect(() => {
+    setToolbarState(getInitialToolbarState());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [recurringRun]);
+
+  useEffect(() => {
+    if (toolbarState) {
+      toolbarState.actions[ButtonKeys.ENABLE_RECURRING_RUN].disabled =
+        recurringRun?.status === V2beta1RecurringRunStatus.ENABLED;
+      toolbarState.actions[ButtonKeys.DISABLE_RECURRING_RUN].disabled =
+        recurringRun?.status !== V2beta1RecurringRunStatus.ENABLED;
+      toolbarState.pageTitle = recurringRun?.display_name!;
+      toolbarState.breadcrumbs = getBreadcrumbs(experiment);
+      updateToolbar(toolbarState);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [toolbarState, experiment, recurringRun]);
+
+  useEffect(() => {
+    refetchRecurringRun();
+  }, [refresh, refetchRecurringRun]);
+
+  const getInitialToolbarState = (): ToolbarProps => {
+    const buttons = new Buttons(props, Refresh);
+    return {
+      actions: buttons
+        .cloneRecurringRun(() => (recurringRun ? [recurringRun.recurring_run_id!] : []), true)
+        .refresh(Refresh)
+        .enableRecurringRun(() => (recurringRun ? recurringRun.recurring_run_id! : ''))
+        .disableRecurringRun(() => (recurringRun ? recurringRun.recurring_run_id! : ''))
+        .delete(
+          () => (recurringRun ? [recurringRun.recurring_run_id!] : []),
+          'recurring run config',
+          deleteCallback,
+          true /* useCurrentResource */,
+        )
+        .getToolbarActionMap(),
+      breadcrumbs: [],
+      pageTitle: '',
+    };
+  };
+
+  const deleteCallback = (_: string[], success: boolean) => {
```

Comment:
Usually the definition of function comes earlier than the calling of this function, can you move `deleteCallback` to be above definition of `getInitialToolbarState`?